### PR TITLE
reduce 250ms to 50ms retry looking for metacache block

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
reduce 250ms to 50ms retry looking for metacache block

## Motivation and Context
the metacache object does not exist until the time
the listing arrives, this leads to arbitrary delay
of 250ms for the first attempt.

There is no real guarantee of when it will be available 
however, we can reduce the overall retry to 50ms to ensure 
the overall response time is not highly skewed in this manner.

```
GET /testbucket/?delimiter=&encoding-type=url&fetch-owner=true&list-type=2&prefix= HTTP/1.1
Response Time:  32.508267ms (direct from disk)

GET /testbucket/?continuation-token=YmFzaC1jb21wbGV0aW9uL2NvbXBsZXRpb25zL2ZpbGVbbWluaW9fY2FjaGU6djIsaWQ6ODM1Y2ZlYTEtYWMzOC00NzZmLWI3NWYtM2RjOTEyMTcwNTc4LHA6MCxzOjBd&delimiter=&encoding-type=url&fetch-owner=true&list-type=2&prefix= HTTP/1.1
Response Time:  280.822262ms (attempt to stream from metacache object)
```

## How to test this PR?
Just create a bunch of files with more than 2000+ entries 
and then perform a recursive listing and observe the 
response times.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by harshavardhana** [comment]
After this change

```
GET /testbucket/?delimiter=&encoding-type=url&fetch-owner=true&list-type=2&prefix= HTTP/1.1
Response Time:  38.627381ms
GET /testbucket/?continuation-token=YmFzaC1jb21wbGV0aW9uL2NvbXBsZXRpb25zL2ZpbGVbbWluaW9fY2FjaGU6djIsaWQ6OWYwZDcyYWMtZGM0MS00ZDFkLThlMzQtZjk2ZTMxMjA1Y2UzLHA6MCxzOjBd&delimiter=&encoding-type=url&fetch-owner=true&list-type=2&prefix= HTTP/1.1
Response Time:  51.404211ms
```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :x: [more...](#aeaee34b-0c14-4ad1-91e4-f8c387f945d0) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#5aad3f4d-8580-44fc-a2e6-4667904043cb) |
| mint-pools.sh | :x: [more...](#1159367f-6b22-44bd-bca1-8bbe3c478108) |
<a name="5aad3f4d-8580-44fc-a2e6-4667904043cb"></a>
### 16795-ee0ab37/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.4:30587
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 880bc9b49bce:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 9 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 10 seconds
(5/14) Running awscli tests ... done in 1 minutes and 16 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 17 seconds
(8/14) Running minio-go tests ... done in 50 seconds
(9/14) Running minio-java tests ... FAILED in 5 minutes and 13 seconds
{
  "name": "minio-java",
  "function": "deleteBucketPolicy()",
  "duration": 307272,
  "status": "FAIL",
  "error": "java.net.SocketTimeoutException: timeout >>> [okio.SocketAsyncTimeout.newTimeoutException(JvmOkio.kt:147), okio.AsyncTimeout.access$newTimeoutException(AsyncTimeout.kt:158), okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:337), okio.RealBufferedSource.indexOf(RealBufferedSource.kt:427), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:320), okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:106), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:79), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(9/14) Running minio-js tests ... done in 51 seconds
(10/14) Running minio-py tests ... done in 2 minutes and 4 seconds
(11/14) Running s3cmd tests ... done in 16 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
<a name="aeaee34b-0c14-4ad1-91e4-f8c387f945d0"></a>
### 16795-ee0ab37/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.7:30908
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 5d49abf32195:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... done in 1 minutes and 14 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 15 seconds
(8/14) Running minio-go tests ... done in 33 seconds
(9/14) Running minio-java tests ... done in 17 seconds
(10/14) Running minio-js tests ... FAILED in 1 minutes and 14 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-43d20165-cc76-4edf-bb8d-cf03f15a37bc, events: s3:ObjectCreated:*",
  "duration": 63419,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 1 minutes and 28 seconds
(11/14) Running s3cmd tests ... done in 16 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 8 seconds

Executed 13 out of 14 tests successfully.
```
<a name="1159367f-6b22-44bd-bca1-8bbe3c478108"></a>
### 16795-ee0ab37/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.3:31739
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 773edf240841:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 9 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 10 seconds
(5/14) Running awscli tests ... done in 1 minutes and 16 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 18 seconds
(8/14) Running minio-go tests ... FAILED in 1 minutes and 55 seconds
{
  "args": {
    "bucketName": "minio-go-test-asi5griv4nyjpes0",
    "objectName": "yfavx22pcdqs534wfmowzo5pud3g53"
  },
  "duration": 64180,
  "error": "read tcp 172.17.0.2:44798->15.15.15.3:31739: read: connection reset by peer",
  "function": "GetObject(bucketName, objectName)",
  "message": "ReadAt failed",
  "name": "minio-go: testSSES3EncryptedGetObjectReadAtFunctional",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... done in 37 seconds
(9/14) Running minio-js tests ... done in 53 seconds
(10/14) Running minio-py tests ... done in 2 minutes and 16 seconds
(11/14) Running s3cmd tests ... done in 1 minutes and 18 seconds
(12/14) Running s3select tests ... done in 5 seconds
(13/14) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-erasure.sh | :x: [more...](#28be3e6c-9ba9-4752-981b-2ce75b6baeb0) |
| mint-pools.sh | :x: [more...](#e6c8aa4a-98fd-4190-8b1d-73706b72c1ad) |
<a name="e6c8aa4a-98fd-4190-8b1d-73706b72c1ad"></a>
### 16795-1c6b645/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:32630
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 19bd164ba2e0:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 1 minutes and 15 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 12 seconds
{
  "name": "mc",
  "duration": "11150",
  "function": "test_rb",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-30839 --quiet --no-color mb myminio/mc-test-bucket-10784-2  >>> \nmc: <ERROR> Unable to make bucket `myminio/mc-test-bucket-10784-2`. Put \"http://15.15.15.1:32630/mc-test-bucket-10784-2/\": dial tcp 15.15.15.1:32630: i/o timeout"
}
(7/14) Running minio-go tests ... done in 1 minutes and 11 seconds
(8/14) Running minio-java tests ... done in 37 seconds
(9/14) Running minio-js tests ... done in 52 seconds
(10/14) Running minio-py tests ... FAILED in 3 minutes and 6 seconds
{
  "name": "minio-py:test_presigned_get_object_default_expiry",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89",
    "object_name": "ad14f40c-51ff-4631-9a4f-344aa094911c"
  },
  "message": "S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89, request_id: 174B750B1F98A053, host_id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, bucket_name: minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1269, in test_presigned_get_object_default_expiry\n    _CLIENT.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89, request_id: 174B750B1F98A053, host_id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, bucket_name: minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89\n",
  "duration": 56972
}
(10/14) Running s3cmd tests ... done in 17 seconds
(11/14) Running s3select tests ... done in 5 seconds
(12/14) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 12 out of 14 tests successfully.
```
<a name="28be3e6c-9ba9-4752-981b-2ce75b6baeb0"></a>
### 16795-1c6b645/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.8:31271
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp c530c85025b3:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 7 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... done in 1 minutes and 14 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 17 seconds
{
  "name": "mc",
  "duration": "12179",
  "function": "test_mv_object",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-12670 --quiet --no-color mv myminio/mc-test-bucket-2927/dir-16081-4512/object-2 /tmp/tmp.IL6tHb6uBF/file  >>> \nmc: <ERROR> Unable to validate source `myminio/mc-test-bucket-2927/dir-16081-4512/object-2`."
}
(7/14) Running minio-go tests ... done in 49 seconds
(8/14) Running minio-java tests ... done in 18 seconds
(9/14) Running minio-js tests ... done in 47 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 28 seconds
(11/14) Running s3cmd tests ... done in 15 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 15 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by klauspost** [comment]
No. You are just increasing chatter by 5x. When this is hit there has already been at least 2 responses sent, and we can expect there to be more (subsequent blocks). 

The subsequent calls will just be able to complete faster, so you are not in most cases not decreasing latency of the total listing.  At least not to a degree that can justify the load increase. The listing completes independently of whatever this latency is.

This PR does not add any value IMO. You can always create a case that would justify lowering this value arbitrarily.

**Comment by harshavardhana** [comment]
> No. You are just increasing chatter by 5x. When this is hit there has already been at least 2 responses sent, and we can expect there to be more (subsequent blocks).
> 
> The subsequent calls will just be able to complete faster, so you are not in most cases not decreasing latency of the total listing. At least not to a degree that can justify the load increase. The listing completes independently of whatever this latency is.
> 
> This PR does not add any value IMO. You can always create a case that would justify lowering this value arbitrarily.

So how do we address the random latency increase? For me this looks like a design bug.

Because we could have simply streamed without persistence at this point and there won't be any arbitrary latency increase.

50ms is completely okay, I would even keep it at 100ms - what I see right now is unnecessary latency added for even faster hardware just because we lack synchronization from list batch sent v/s continuation-token.

This cost is real and easily gets added for spark like workloads. Either this PR is the solution or we need a proper solution for this where we don't have to arbitrarily retry looking for metacache object.

**Comment by klauspost** [comment]
The method is unsynchronized and unlocked to avoid needless chatter.

On a loaded system you will have requests waiting for minutes for a block to complete. You are just increasing the non-productive workload on these systems with a factor 5.

If you want to decrease the latency, make it block on `checkMetacacheState`, but then you would have a long running request waiting to get notified for the request.

> Because we could have simply streamed without persistence at this point and there won't be any arbitrary latency increase.

No. The listing is running faster and doesn't have the inefficiencies of stateless requests.

The first request stopped listing once it had served the first request. The second request restarted the listing and returned the results once it had enough results. The second listing is now running async and persisting the results. 

We are waiting for the first block of 5000 object entries to be saved.

> Either this PR is the solution or we need a proper solution for this where we don't have to arbitrarily retry looking for metacache object.

You are not getting any useful increase in speed.  You are getting the next 5 responses instantly, as compared to having to wait for each. A random request having to wait 250ms cannot possibly have any real impact.

**Comment by klauspost** [comment]
@harshavardhana OK. Compromise:

First wait 50ms, if that fails, wait 250ms for every subsequent attempt.

**Comment by harshavardhana** [comment]
@klauspost done PTAL

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/metacache-set.go [hunk 1]
```diff
@@ -440,11 +444,18 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
...
+				if retries == 1 {
+					time.Sleep(retryDelayFirst)
+				} else {
+					time.Sleep(retryDelay)
+				}
 				continue
 			case InsufficientReadQuorum:
-				retries++
-				time.Sleep(retryDelay)
+				if retries == 1 {
```
Comment (kannappanr) [inline cmd/metacache-set.go:455]: why is retries not getting incremented?

Comment (harshavardhana) [reply cmd/metacache-set.go:455]: That's a mistake we need to keep that.

Comment (klauspost) [reply cmd/metacache-set.go:455]: Yes, please

**Comment by harshavardhana** [comment]
I did it slightly differently, thought we can keep max retries up to 5 and then keep `retries * retryDelay` while keeping the default retry delay to be 50ms

- 50ms (first retry)
- 100ms (second retry)
- 150ms (third retry)
- 200ms (fourth retry)
- 250ms (fifth retry)

**Comment by klauspost** [comment]
@harshavardhana That was not the deal we made. 50 -> 250.

**Comment by harshavardhana** [comment]
> @harshavardhana That was not the deal we made. 50 -> 250.

Correct I wanted to see if this is also an acceptable variant.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :x: [more...](#207fd13a-ad45-47a2-9df4-20d41cf3a8bf) |
<a name="207fd13a-ad45-47a2-9df4-20d41cf3a8bf"></a>
### 16795-c0311c4/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.5:30329
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 7b40f1b287af:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 9 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 50 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 1 minutes and 16 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 18 seconds
(8/14) Running minio-go tests ... done in 58 seconds
(9/14) Running minio-java tests ... done in 37 seconds
(10/14) Running minio-js tests ... done in 52 seconds
(11/14) Running minio-py tests ... FAILED in 2 minutes and 12 seconds
{
  "name": "minio-py:test_stat_object",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-17c0f243-b8f9-440f-82e7-33972ea176d7",
    "object_name": "71beef91-5f4a-400b-b165-db59e55d6ac5-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "X-Amz-Meta-Testing": "value"
    },
    "content_type": "application/octet-stream"
  },
  "message": "('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 449, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 3, in raise_from\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 444, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.8/http/client.py\", line 1348, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.8/http/client.py\", line 316, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.8/http/client.py\", line 277, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n  File \"/usr/lib/python3.8/socket.py\", line 669, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 820, in test_stat_object\n    _test_stat_object(log_entry, sse)\n  File \"/mint/run/core/minio-py/tests.py\", line 794, in _test_stat_object\n    result = _CLIENT.put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1766, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1725, in put_object\n    upload_id = self._create_multipart_upload(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1565, in _create_multipart_upload\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 550, in increment\n    raise six.reraise(type(error), error, _stacktrace)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/packages/six.py\", line 769, in reraise\n    raise value.with_traceback(tb)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 449, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 3, in raise_from\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 444, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.8/http/client.py\", line 1348, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.8/http/client.py\", line 316, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.8/http/client.py\", line 277, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n  File \"/usr/lib/python3.8/socket.py\", line 669, in readinto\n    return self._sock.recv_into(b)\nurllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))\n",
  "duration": 114850
}
(11/14) Running s3cmd tests ... done in 18 seconds
(12/14) Running s3select tests ... done in 5 seconds
(13/14) Running versioning tests ... done in 3 minutes and 32 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
> > @harshavardhana That was not the deal we made. 50 -> 250.
> 
> Correct I wanted to see if this is also an acceptable variant.

I can revert this back @klauspost if you insist and move back to 50 -> 250 direct jump.

**Comment by klauspost** [comment]
> I can revert this back if you insist and move back to 50 -> 250 direct jump.

Yes, then you get the 👍🏼  :)

A progressive rampup does nothing. If it isn't available after 50ms, then we might as well wait longer and not stress the system.

**Comment by harshavardhana** [comment]
@klauspost its reverted back to simply "retries == 1 then sleep for 50ms" approach.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :x: [more...](#79b4ef89-a458-44a8-909a-430fdb570a35) |
<a name="79b4ef89-a458-44a8-909a-430fdb570a35"></a>
### 16795-77750de/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:31100
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp aea79905e5e0:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 9 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... FAILED in 5 minutes and 10 seconds
{"name": "awscli", "duration": 307417, "function": "make_bucket\n", "status": "FAIL", "error": "Connect timeout on endpoint URL: "http://15.15.15.1:31100/awscli-mint-test-bucket-17701""}
(5/14) Running healthcheck tests ... FAILED in 30 seconds
{
  "alert": "",
  "args": null,
  "duration": 30029,
  "error": "Get \"http://15.15.15.1:31100/minio/health/live\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(5/14) Running mc tests ... FAILED in 20 seconds
{
  "alert": "",
  "args": null,
  "duration": 30029,
  "error": "Get \"http://15.15.15.1:31100/minio/health/live\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(5/14) Running minio-go tests ... done in 1 minutes and 28 seconds
(6/14) Running minio-java tests ... done in 37 seconds
(7/14) Running minio-js tests ... done in 52 seconds
(8/14) Running minio-py tests ... done in 7 minutes and 28 seconds
(9/14) Running s3cmd tests ... done in 17 seconds
(10/14) Running s3select tests ... done in 5 seconds
(11/14) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 11 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]

**Pull request merged**